woocommerce/woocommerce
# Add `orders/statuses` endpoint to the REST API

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR aimes to introduces a missing endpoint to the REST API to list available order statuses. It add the `wp-json/wc/store/v1/orders/statuses` endpoint.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch
2. Visit http://localhost:8888/wp-json/wc/v3/orders/statuses or test via cURL with `curl -H "Accept: application/json" http://localhost:8888/wp-json/wc/v3/orders/statuses`
3. You might need to run `composer dump-autoload` if cURL doesn't return anything

You should get this result:

<img width="293" alt="image" src="https://github.com/user-attachments/assets/c7590a08-e2f7-432b-9056-73c04c4879d9">




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
